chain/consensus/hierarchical/subnet/submgr/manager.go

```diff
@@ -389,6 +389,9 @@ func (s *Service) GetSubnetState(ctx context.Context, id address.SubnetID, actor
 }
 
 func (s *Service) AddSubnet(ctx context.Context, params *hierarchical.SubnetParams) (address.Address, error) {
+	s.lk.Lock()
```

Comment:
Do you mind removing the TODOs in the rest of the subnet manager functions where we use `s.lk`? I don't think we need to re-thing the locking strategy, at least for now.

Reply:
removed